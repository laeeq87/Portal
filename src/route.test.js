import React from "react";
import { render, cleanup} from '@testing-library/react';
import Routes from "./routes";
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import Home from "./home";
import {
    Route,
    Switch,Router,
    Redirect,
  } from "react-router-dom";


afterEach(()=> {
    cleanup();
});

const props = {
    match:{
        params: jest.fn(),
    }
  }

it("Should Render Porfile Component", ()=> {
    const history = createMemoryHistory(['/', '/login', '/account']);
    const { debug } =render(
        <Router history={history}>
            <Switch>
                <Route path="/account/:id" exact component={Home}/>
                <Route path="/account" exact><Redirect to="/account/profile" /><div>Account profile Rendered</div></Route>
            </Switch>
        </Router>
    );
    debug()
}); 


test('matches snapshot', ()=> {
    const history = createMemoryHistory(['/', '/login', '/account']);
    const tree = renderer.create(
        <Router history={history}>
            <Routes{...props} history={history}/>
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})
