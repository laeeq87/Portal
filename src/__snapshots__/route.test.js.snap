// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <div
      className="row justify-content-center align-items-cente"
    >
      <div
        className="col-6 text-center my-5 my-border"
      >
        <div
          className="row justify-content-center my-5 align-items-cente"
        >
          <div
            className="col-md-12 text-center "
          >
            <h3>
               Welcome to our Portal
            </h3>
          </div>
          <div
            className="col-md-8 text-center mt-4"
          >
            <h5>
              Please Go to Login :
            </h5>
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
               Login 
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
